docs/source/user_guides/slot_extraction.rst

```diff
@@ -0,0 +1,98 @@
+Slot Extraction
+---------------
+
+Introduction
+~~~~~~~~~~~~
+
+Extracting and filling slots is an essential part of any conversational service
+that comprises the inherent business logic. Like most frameworks, DFF
+provides components that address this task as a part of its `slots` module.
+These can be easily customized to leverage neural networks specifically designed
+for slot extraction or any other logic you might want to integrate.
+
+API overview
+~~~~~~~~~~~~
+
+The basic building block of the API is the `Slot` class and its descendants
+that vary depending on the value extraction logic. Each slot has a name
+by which it can be accessed and a method for extracting values.
+Below, we demonstrate the most basic class that extracts values
+from user utterances using a regular expression.
+
+.. code-block:: python
+
+    from dff.script.slots import RegexpSlot
+    ...
+    email_slot = RegexpSlot(name="email", regexp=r"^[A-Z][a-z]+?(?= )")
+
+Individual slots can be grouped allowing the developer to access them together
+as a namespace. This can be achieved using the `GroupSlot` component
+that can be initialized with a list of other slot instances as its children.
+The group slots also allow for arbitrary nesting, e. g. it is possible to include
+group slots in other group slots.
+
+.. code-block:: python
+
+    from dff.script.slots import GroupSlot
+    ...
+    profile_slot = slots.GroupSlot(name="profile", children=[email_slot])
+
+When all slots have been defined, you can refer to them in your script.
+Slots can be extracted at the `PRE_TRANSITION_PROCESSING` stage, the appropriate
+function being provided by the `processing` submodule.
+The only argument is the list of names of the slots that you want to extract
+from user utterances.
+
+.. code-block:: python
+
+    from dff.script.slots.processing import extract
+    ...
+    PRE_TRANSITIONS_PROCESSING: {"extract_action": extract(["slot_name"])}
```

Comment:
We can access the name strings using the `name` attribute on slots which helps us make the most of syntax highlighting.
E.g we don't have to do
```
slot = Slot()
extract([slot])
```
Since we are now able to do
```
slot = Slot()
extract([slot.name])
```

Reply:
Ok